src/routeApp/tests/controllers.test.js

```diff
@@ -0,0 +1,157 @@
+import bcrypt from 'bcrypt';
+import chai, { expect } from 'chai';
+import chaiHttp from 'chai-http';
+import app from '../../app.js';
+import logger from '../../configs/winston.js';
+import DataSource from '../../data-source.js';
+import User from '../../models/user.js';
+
+chai.use(chaiHttp);
+chai.should();
+
+describe('Route router tests', () => {
+	before(async () => {
+		try {
+			await DataSource.query('TRUNCATE "Route" CASCADE');
+			await DataSource.query('TRUNCATE "User" CASCADE');
+		} catch (error) {
+			logger.error(error.message);
+		}
+	});
+
+	let routeEdit;
+	let token;
+	let routeData;
+
+	it('should login an admin user', async () => {
+		const password = await bcrypt.hash('Andela@1234', 5);
+		const newUser = User.create({
+			email: 'test123@gmail.com',
+			password,
+			firstName: 'test',
+			lastName: 'Andela',
+			role: 'admin',
+		});
+		const user = await newUser.save();
+		const info = { email: user.email, password: 'Andela@1234' };
+		const response = await chai
+			.request(app)
+			.post('/api/v1/accounts/login')
+			.send(info);
+		expect(response).to.have.status(200);
+		token = response.body.data.access_token;
+	});
+
+	it('should get response with no Routes', async () => {
+		const res = await chai
+			.request(app)
+			.get('/api/v1/routes')
+			.set('Authorization', `Bearer ${token}`);
+		expect(res).to.have.status(200);
+		expect(res.body.data).to.have.length(0);
+	});
+
+	it('should create a new Route', async () => {
+		routeData = {
+			origin: 'Nyanza',
+			destination: 'Nyabugogo',
+			distance: 146,
+		};
+		const res = await chai
+			.request(app)
+			.post('/api/v1/routes')
+			.send(routeData)
+			.set('Authorization', `Bearer ${token}`);
+		expect(res).to.have.status(201);
+		routeEdit = res.body.data.id;
+	});
+
+	it('should not create a new Route that already exist', async () => {
+		const res = await chai
+			.request(app)
+			.post('/api/v1/routes')
+			.send(routeData)
+			.set('Authorization', `Bearer ${token}`);
+		expect(res).to.have.status(409);
+	});
+
+	it('should get all Routes', async () => {
+		const res = await chai
+			.request(app)
+			.get('/api/v1/routes')
+			.set('Authorization', `Bearer ${token}`);
+		expect(res).to.have.status(200);
+	});
+
+	it('should get single Route with Id', async () => {
+		const response = await chai
+			.request(app)
+			.get(`/api/v1/routes/${routeEdit}`)
+			.set('Authorization', `Bearer ${token}`);
+		expect(response).to.have.status(200);
+		// console.log(response.body);
+	});
+	it('should Edit Route', async () => {
+		const editRoute = {
+			origin: 'Nyanza',
+			destination: 'Remera',
+			distance: 146,
+		};
```

Comment:
make this editRoute as global variable or toast it in the dummy data and call it from there

Reply:
Well done